smu/geometry/bond_lengths.py

```diff
@@ -35,6 +34,11 @@ def process(self, conformer):
     bonded = utilities.bonded(bt)
 
     natoms = len(bt.atoms)
+    if len(geom.atom_positions) != natoms:
+      return
+
+    if conformer.fate != dataset_pb2.Conformer.FATE_SUCCESS:
```

Comment:
I'd like to do this geometry detection for all fates, NOT just FATE_SUCCESS. We can always pick a subset of things later on.

Reply:
I am running some of these calculations now. I imagine most of the disassociated molecules will fail, since they will have a bond that is outside the range of what is acceptable. 
That will be an interesting check.